### PR TITLE
Switch to glide for vendoring & bump k8s v1.62 & client-go v3.0.0-beta.0

### PR DESCRIPTION
**Pull request by aaronlevy**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,34 +53,9 @@ conformance-%: clean all
 	@sleep 30 # Give addons a little time to start
 	@cd hack/$*-node && ./conformance-test.sh
 
-#TODO(aaron): the k8s.io/client-go upstream package is a symlink with relative path. Making note because we change symlink path.
-# This will naively try and create a vendor dir from a k8s release
-# USE: make vendor VENDOR_VERSION=vX.Y.Z
-
-#TODO(aaron): Remove etcd-operator-vendor.patch once etcd-operator bumps client-go version
```
Comment (diegs) [inline Makefile]: Did you also remove etcd-operator-vendor.patch?

File: Makefile [hunk 2]
```diff
@@ -53,34 +53,9 @@ conformance-%: clean all
...
-	@cd $@/k8s.io/ && ln -sf kubernetes/staging/src/k8s.io/apiserver apiserver
-	@rm -rf $@/k8s.io/kubernetes/vendor $@/k8s.io/kubernetes/.git
-	@echo "vendoring etcd-operator spec: $(ETCD_OPERATOR_VERSION)"
-	@git clone https://github.com/coreos/etcd-operator /tmp/etcd-operator > /dev/null 2>&1
-	@cd /tmp/etcd-operator && git checkout $(ETCD_OPERATOR_VERSION) > /dev/null 2>&1
-	@mkdir -p $@/github.com/coreos/etcd-operator/pkg/
-	@cp -r /tmp/etcd-operator/pkg/spec $@/github.com/coreos/etcd-operator/pkg/
-	@rm -rf /tmp/etcd-operator
-	@git apply etcd-operator-vendor.patch
+	@glide update --strip-vendor
```
Comment (diegs) [inline Makefile]: Maybe add quick note on glide and glide-vc dependencies to the readme?

File: Makefile (hunk 1 above)
Comment (aaronlevy) [inline Makefile]: woops, forgot that - removed now.

File: Makefile (hunk 2 above)
Comment (aaronlevy) [inline Makefile]: Just updated the release documentation if you can take a look

Comment (dghubble) [inline Makefile]: If you can get away with it, several projects are using @glide-vc --use-lock-file --no-tests --only-code

**Pull request merged**